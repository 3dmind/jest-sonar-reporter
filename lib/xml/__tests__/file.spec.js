'use strict'

const xml = require('xml')
const file = require('../file')
const path = require('path')

describe('file', () => {
  test('<file path=""></file>', () => {
    const mock = {
      testFilePath: 'test/FooTest.js',
      testResults: []
    }

    const actualReport = xml(file(mock, {useRelativePath: false}))

    expect(actualReport).toMatchSnapshot()
  })

  test('path value is relative', () => {
    const mock = {
      testFilePath: path.join(process.cwd(), 'test', 'RelativePathTest.js'),
      testResults: []
    }

    const actualReport = xml(file(mock, {useRelativePath: true}))

    expect(actualReport).toMatchSnapshot()
  })

  test('testCase tag', () => {
    const mock = {
      testFilePath: 'test/FooTest.js',
      testResults: [
        {title: 'lorem ipsum'},
        {title: 'lorem ipsum'}
      ]
    }

    const actualReport = xml(file(mock, {useRelativePath: false}), true)

    expect(actualReport).toMatchSnapshot()
  })
})
