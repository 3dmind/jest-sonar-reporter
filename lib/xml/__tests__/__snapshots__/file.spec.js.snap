// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file <file path=""></file> 1`] = `"<file path=\\"test/FooTest.js\\"></file>"`;

exports[`file path value is relative 1`] = `"<file path=\\"test/RelativePathTest.js\\"></file>"`;

exports[`file testCase tag 1`] = `
"<file path=\\"test/FooTest.js\\">
    <testCase name=\\"lorem ipsum\\" duration=\\"0\\"/>
    <testCase name=\\"lorem ipsum\\" duration=\\"0\\"/>
</file>"
`;
